src/component/parser/json/converter/CollaborationConverter.ts

```diff
@@ -0,0 +1,39 @@
+import { JsonConvert, JsonConverter, OperationMode, ValueCheckingMode } from 'json2typescript';
+import { AbstractConverter, ensureIsArray } from './AbstractConverter';
+import ShapeBpmnElement from '../../../../model/bpmn/shape/ShapeBpmnElement';
+import { ShapeBpmnElementKind } from '../../../../model/bpmn/shape/ShapeBpmnElementKind';
+import { Collaboration, Semantic } from '../Definitions';
+
+const convertedPoolBpmnElements: ShapeBpmnElement[] = [];
+
+export function findPoolBpmnElement(id: string): ShapeBpmnElement {
+  return convertedPoolBpmnElements.find(i => i.id === id);
+}
+
+// TODO : To move in a singleton object to use here and in the BpmnJsonParser
+const jsonConvert: JsonConvert = new JsonConvert();
+jsonConvert.operationMode = OperationMode.ENABLE;
+jsonConvert.ignorePrimitiveChecks = false; // don't allow assigning number to string etc.
+jsonConvert.valueCheckingMode = ValueCheckingMode.DISALLOW_NULL; // never allow null
+//////////////////////////////////////////////////////////////
+
+@JsonConverter
+export default class CollaborationConverter extends AbstractConverter<Collaboration> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  deserialize(collaboration: any): Collaboration {
+    try {
+      // Deletes everything in the array, which does hit other references. For better performance.
+      convertedPoolBpmnElements.length = 0;
+
+      ensureIsArray(collaboration['participant']).forEach(participant => {
+        const poolShape = new ShapeBpmnElement(participant.id, participant.name, ShapeBpmnElementKind.POOL);
```

Comment:
Be careful. All participants are not Process/Pool. You need to verify if the participant has the field 'processRef'.

Reply:
I think it's easier to use the processRef as id of the pool, and when we parse the lane, add the id of the process as parentId of the lane.
No need to parse again the child of the pool like we do for the lane ;)

Reply:
> Be careful. All participants are not Process/Pool. You need to verify if the participant has the field 'processRef'.

good catch, I will add a non process participant in the test to validate that

> I think it's easier to use the processRef as id of the pool, and when we parse the lane, add the id of the process as parentId of the lane.
> No need to parse again the child of the pool like we do for the lane ;)

yes, pretty clever, this will solve the issue I currently have !